CMakeLists.txt

```diff
@@ -158,6 +159,48 @@ include(cmake/leveldb.cmake)
 include(cmake/minisketch.cmake)
 include(cmake/secp256k1.cmake)
 
+include(ProcessConfigurations)
+set_default_config(RelWithDebInfo)
+
+# Redefine configuration flags.
+add_compile_definitions($<$<CONFIG:Debug>:DEBUG>)
+add_compile_definitions($<$<CONFIG:Debug>:DEBUG_LOCKORDER>)
+add_compile_definitions($<$<CONFIG:Debug>:DEBUG_LOCKCONTENTION>)
+add_compile_definitions($<$<CONFIG:Debug>:RPC_DOC_CHECK>)
+add_compile_definitions($<$<CONFIG:Debug>:ABORT_ON_FAILED_ASSUME>)
+# We leave assertions on.
+if(MSVC)
+  remove_c_flag_from_all_configs(/DNDEBUG)
+  remove_cxx_flag_from_all_configs(/DNDEBUG)
+else()
+  remove_c_flag_from_all_configs(-DNDEBUG)
+  remove_cxx_flag_from_all_configs(-DNDEBUG)
+
+  # Prefer -O2 optimization level. (-O3 is CMake's default for Release for many compilers.)
+  replace_c_flag_in_config(Release -O3 -O2)
+  replace_cxx_flag_in_config(Release -O3 -O2)
+
+  set(debug_c_flags "")
+  set(debug_cxx_flags "")
+  try_append_cxx_flags(debug_cxx_flags "-O0" RESULT_VAR compiler_supports_O0)
+  if(compiler_supports_O0)
+    string(STRIP "${debug_c_flags} -O0" debug_c_flags)
+  endif()
+  try_append_cxx_flags(debug_cxx_flags "-g3" RESULT_VAR compiler_supports_g3)
```

Comment:
The "-g" flags should be on for Debug + RelWithDebInfo, off for MinSizeRel/Release. But it appears they're applied here unconditionally?

Reply:
Hmm, though your multi-config output looks as I'd expect. I must be misunderstanding something. Ignore the above comment for now.